### DISCUSSION TITLE
Can't Reach Database Server (P1001)

### DISCUSSION THREAD

**Discussion by Wael-Dghim** [prisma/prisma#23985] [Q&A] [answered]
### Question

I have a domain name purchased from a certain hosting provider, and with it I get access to 2 MySQL databases which are fully functional. I can see the database information from cPanel and visit phpMyAdmin to see the schemas, data and connection information.

I'm just running a test to migrate the models from my schema to the database from Prisma ORM. I get a (P1001) error saying :

> P1001: Can't reach database server at `ip`:`port`
Please make sure your database server is running at `ip`:`port`.

My code : 
**schema.prisma**

```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

**.env**

`DATABASE_URL="mysql://xuser:xpassword@xhost:xport/xdb"`

When i ping the port, I get a response but the issue lies with PrismaORM trying to connect to the database

### Expected behavior (optional)

It should migrate my example models to the database that is hosted on cPanel.

### Information about Prisma Schema, Client Queries and Environment (optional)

**schema.prisma**
```
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

**.env**

`DATABASE_URL="mysql://xuser:xpassword@xhost:xport/xdb"`

**Comment by enomis125** [comment]
I'm not sure if it'll resolve the problem or if you have already did this, but when I was using cPanel and a mysql database I went to the remote mysql and inserted my ip to be able to make a connection.

![image](https://github.com/prisma/prisma/assets/116946584/69173c79-4062-49c0-b9e7-b1f2271fe35d)

![image](https://github.com/prisma/prisma/assets/116946584/c0dca82c-1ce6-41ac-921f-81c35f315111)

**Reply by Wael-Dghim** [reply]
Yes, thank you for mentioning that. I actually did the config for remote MySQL, I tried with current IP, but since my IP is volatile and changes randomly, i used the wild card (%) and (%.%.%.%) just to be sure.

Still nothing :disappointed:

**Comment by ludralph** [comment]
Hi @Wael-Dghim 👋 

Can you confirm that with your connection string, you can connect to your MySQL database using a Database tool like TablePlus or MySQL Workbench?

**Reply by Wael-Dghim** [reply]
I tried using the terminal "mysql --port=... --host=... ..." command to access the database server and I get the same response.

After setting up a connection through MySQL Workbench this is what i get : 
![image](https://github.com/prisma/prisma/assets/104037074/6aeffa48-d03d-430f-b966-e592a82b518e)

I'm not entirely sure this is a prisma issue anymore but I would really appreciate the help!

**Comment by Wael-Dghim** [comment, marked as answer]
For anyone that faces a similar issue, with cPanel... I contacted the technical support team of my hosting provider and they said it was a security measure and that their service doesn't allow for any remote access 🤝.
